src/check_igvm_locked.py

```
@@ -48,33 +69,53 @@ def main():
 
     hosts_not_locked = []
     hosts_locked = []
+
     for host in hosts:
         if not host['igvm_locked']:
             hosts_not_locked.append(host)
             continue
 
         locked_time = datetime.now(timezone.utc) - host['igvm_locked']
-        if locked_time > timedelta(minutes=max_minutes):
+        if locked_time >= timedelta(minutes=max_minutes):
             hosts_locked.append(host)
         else:
             hosts_not_locked.append(host)
-    if args.v:
-        console_out(hosts_locked, max_minutes)
-    nsca_out = nagios_create(hosts_locked, hosts_not_locked, max_minutes)
-    nagios_send(master, nsca_out)
 
+    console_out(hosts_locked, max_minutes)
+
+    results = nagios_create(hosts_locked, hosts_not_locked, max_minutes)
 
-def nagios_create(hosts_locked, hosts_not_locked, max_minutes):
     nsca_output = ""
+    exit_code = 0
+    for result in results:
+        if (len(nsca_output) + len(result)) >= 5000:
```

Somehow that looks like we rather want to achieve something different.
For what exactly are you doing the check for. What do you want to achieve with it?

---

The send_nsca has a limit on it's stdin buffer of roughly 5100 bytes. 
If we don't break it down in chunks, we end up with a string with around 300k characters, and some of the data simply gets silently dropped by send_nsca.